src/themes/default/pages/Product.vue

```diff
@@ -435,8 +440,29 @@ export default {
       } finally {
         this.isStockInfoLoading = false
       }
+    },
+    hasScrolled () {
+      this.scrollCart = window.scrollY
+      if (this.scrollCart > this.lastScrollCart) {
+        this.scrollingAddToCart = true
+      } else if (this.scrollCart === 0) {
+        this.scrollingAddToCart = false
+      }
+      this.lastScrollCart = this.scrollCart
     }
   },
+  beforeMount () {
+    window.addEventListener('scroll', () => {
```

Comment:
```suggestion
    window.addEventListener('scroll', debounce(() => {
      if (this.isScrolling) {
        this.hasScrolled()
        this.isScrolling = false
      }
    }, 250), { passive: true })
```

debounce from lodash instead of interval, and you can put this code inside of event

Reply:
also please call `removeEventListener` in `beforeDestroy` method